File: Samples/iOS/Objective-C/IOS-SDK-Demo-App/Ad Formats/LOMRECViewController.m
//
//  LOMRECViewController.m
//  test-objc-fw-integration
//
//  Created by Clarke Bishop on 3/31/23.
//

/**
 IMPORTANT: Please review and set the placementID inside the Info.plist file. To do so, please do the following:
 
 Navigate to project hierarchy in Xcode, and select Info. Expand Liftoff PlacementID and be sure that MREC is set correctly. Please do the same for Liftoff Dashboard AppID as well inside the Info.plist file.
 */

#import "LOMRECViewController.h"
#import <VungleAdsSDK/VungleAdsSDK.h>

@interface LOMRECViewController () <VungleBannerDelegate>
@property (nonatomic, strong) VungleBanner *mrecAd;
@end

@implementation LOMRECViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  self.navigationItem.title = @"Liftoff Medium Rectangle Ad";
}

#pragma mark - IBAction Methods

- (IBAction)btnPressed:(id)sender {
  if (sender == self.loadBtn && self.placementId != nil) {
    [self loadMrecAd];
  } else if (sender == self.playBtn && [self.mrecAd canPlayAd]) {
    [self playMrecAd];
  } else if (sender == self.closeBtn) {
    [self closeMrecAd];
  }
}

#pragma mark - Private Methods

- (void)loadMrecAd {
  // clean up before load
  if (self.mrecAd != nil) {
    self.mrecAd.delegate = nil;
    self.mrecAd = nil;
    [self resetCallbackLogs];
  }
  self.mrecAd = [[VungleBanner alloc] initWithPlacementId:self.placementId size:BannerSizeMrec];
  self.mrecAd.delegate = self;
  [self.mrecAd load:nil];
}

- (void)playMrecAd {
  [self.mrecAd presentOn:self.bannerAdContainer];
}

- (void)closeMrecAd {
  for (id subview in self.bannerAdContainer.subviews) {
    [subview removeFromSuperview];
  }
}

@end

File: Samples/iOS/Swift/IOS-SDK-Demo-App/Ad Formats/LOMRECViewController.swift
//
//  LOMRECViewController.swift
//  PublicSampleAppTrial
//
//  Created by John Mai on 1/12/23.
//

/**
 IMPORTANT: Please review and set the placementID inside the Info.plist file. To do so, please do the following:
 
 Navigate to project hierarchy in Xcode, and select Info. Expand Liftoff PlacementID and be sure that MREC is set correctly. Please do the same for Liftoff Dashboard AppID as well inside the Info.plist file.
 */

import UIKit
import VungleAdsSDK

class LOMRECViewController: LOBannerViewController {
    private var mrecAd: VungleBanner?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationItem.title = "Liftoff Medium Rectangle Ad"
    }
    
    override func btnPressed(_ sender: UIButton) {
        if sender == self.loadBtn {
            guard let placementId = self.placementId else {
                return
            }
            self.mrecAd = VungleBanner(placementId: placementId, size: BannerSize.mrec)
            self.mrecAd?.delegate = self
            self.mrecAd?.load()
        } else if sender == self.playBtn {
            // Determines whether the banner object is nil in addition to whether the ad assets have been downloaded successfully
            if ((self.mrecAd?.canPlayAd()) != nil) {
                self.mrecAd?.present(on: self.bannerAdContainer)
            }
        } else if sender == self.closeBtn {
            for subViews in self.bannerAdContainer.subviews {
                subViews.removeFromSuperview()
            }
            self.mrecAd?.delegate = nil
            self.mrecAd = nil
            
            AppUtil.resetLogMessage(tableView: self.tableView, callbacks: self.callbackLogs)
        }
    }
}
